### PR TITLE
Enhance documentation for Snake Tutorial functions

### PR DESCRIPTION
**Pull request by satendrapalsingh**

### CODE REVIEW

File: docs/docs/functions/macros.md [hunk 1]
```diff
@@ -39,20 +57,31 @@ Defines new blocks that can be inserted into an existing chain.
...
+    ```clojure linenums="1"
+    ;; def + Chain
     (def my-chain
     (Chain "my-chain"
         ;; blocks here
     ))
     ```
 
-For a chain to be executed, it must first be scheduled on a `node` and then that that node needs to run.
+A chain is a stateful function.
```
Comment (sinkingsugar) [inline docs/docs/functions/macros.md:80]: this is a good statement but not in the right place and maybe needs to be continued?

Comment (Kryptos-FR) [reply docs/docs/functions/macros.md:80]: I think we will later have a link to a more in-depth article about chain and blocks here.
We could already set that link and point to a "TODO" or "coming soon" placeholder page.

**Review by Kryptos-FR — APPROVED** [review]

File: docs/docs/functions/macros.md [hunk 2]
```diff
@@ -143,8 +143,126 @@ Defines a new function.
...
 
     ```clojure linenums="1"
+    (defn fn-name [fn-parms] ;; function-name followed by input parameters in '[]'
+        (Msg parm)           ;; function body that returns a value
+    )
+    ```
+
+The function definition consists of the function name followed by its input parameters (parms), in `[]`.
+
+If there are no input parms the `[]` remains empty. Multiple input parms may be passed as a sequence.
```
Comment (Kryptos-FR) [inline docs/docs/functions/macros.md:153]: I would keep either "parameters" full word or use `fn-params` in "code" formatting. But abbreviations in text form should be avoided in my opinion.

File: docs/docs/functions/macros.md [hunk 3]
```diff
@@ -143,8 +143,126 @@ Defines a new function.
 === "Code"
 
     ```clojure linenums="1"
+    (defn fn-name [fn-parms] ;; function-name followed by input parameters in '[]'
+        (Msg parm)           ;; function body that returns a value
+    )
+    ```
+
+The function definition consists of the function name followed by its input parameters (parms), in `[]`.
```
Comment (Kryptos-FR) [inline docs/docs/functions/macros.md:151]:
```suggestion
The function definition consists of the function name followed by its input parameters (`fn-params`), in `[]`.
```

File: docs/docs/functions/macros.md [hunk 4]
```diff
@@ -143,8 +143,126 @@ Defines a new function.
 === "Code"
 
     ```clojure linenums="1"
+    (defn fn-name [fn-parms] ;; function-name followed by input parameters in '[]'
+        (Msg parm)           ;; function body that returns a value
```
Comment (Kryptos-FR) [inline docs/docs/functions/macros.md:147]: Usually `params` is the chosen abbreviation for parameters.

```suggestion
    (defn fn-name [fn-params] ;; function-name followed by input parameters in '[]'
        (Msg param)           ;; function body that returns a value
```

File: docs/docs/functions/macros.md [hunk 5]
```diff
@@ -143,8 +143,126 @@ Defines a new function.
...
+A function can be invoked by calling it by name and passing its required parms.
+
+Function with no input parms:
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (defnode main)    
+    (defn func []
+        (Msg "I got no parms"))         ;; prints string text to screen
```
Comment (Kryptos-FR) [inline docs/docs/functions/macros.md:166]: parameters

(same in lines below)

File: docs/docs/functions/macros.md [hunk 6]
```diff
@@ -143,8 +143,126 @@ Defines a new function.
...
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (defnode main)        
+    (defn letslog [name]
+    (->                             ;; defblocks replaced with `->`
+        (Msg "name is:")            ;; multiple blocks can now be written down sequentially
+        (Msg name)
+    )
```
Comment (Kryptos-FR) [inline docs/docs/functions/macros.md:250]: fix indent

**Review by Kryptos-FR — CHANGES_REQUESTED** [review]

File: docs/docs/functions/macros.md (hunk 2 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: Agreed, fixed.

File: docs/docs/functions/macros.md (hunk 4 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: Agreed, fixed.

File: docs/docs/functions/macros.md (hunk 6 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: Agreed, fixed.

File: docs/docs/functions/macros.md (hunk 5 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: Agreed, fixed.

File: docs/docs/functions/macros.md (hunk 3 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: Agreed, fixed.

File: docs/docs/functions/macros.md (hunk 1 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: I agree we should elaborate on this. IMHO this should go in both the 'Chainblocks Architecture' document as well as the 'Chainblocks Primer' document. Both should talk about this in slightly different contexts (the former in terms of blockchain elements, the latter in terms of programming).
I have added the expected links to this page. The links are currently broken but I have made a note to modify these links accordingly when I put content in the chainblocks architecture and primer docs (next in my queue).

**Review by Kryptos-FR — APPROVED** [review]

File: docs/docs/functions/macros.md (hunk 1 above)
Comment (satendrapalsingh) [reply docs/docs/functions/macros.md]: Added TODO pages for :
> Chainblocks architecture guide with [this commit](https://github.com/fragcolor-xyz/chainblocks/pull/238/commits/2db79b473ee45e88eeebd10a624e779c4616b6f1)
> Chainblocks primer doc with [this commit](https://github.com/fragcolor-xyz/chainblocks/pull/238/commits/c595e1fb6692c4d86b3e92096a39648a6307d516)

**Pull request merged**